### DISCUSSION TITLE
Duck typing and create statements

### DISCUSSION THREAD

**Discussion by MrGigabit** [prisma/prisma#25806] [Q&A] [answered]
### Question

If I define a table in prims that has two field, but try to write an object with more fields, nothing complains at compile time because of duck type. For example, if the table is shaped like this 

`{a: number, b: number}`

and I run prisma.create with a record like 

` {a: 1, b: 2, c: "3"}`

Nothing is wrong with this from a compilation standpoint because of duck-typing. However, at run time, this will throw a sql error because it will try to insert the entire record, and sql will complain that column `c` does not exist. 

Is there a way to not run the insert statement on extraneous fields?

**Comment by ludralph** [comment, marked as answer]
Hi @MrGigabit 👋 

Currently, Prisma does not have a built-in feature to automatically ignore extraneous fields during insert operations. Prisma performs runtime validation and will throw an error if you try to insert fields that are not defined in your schema. There is an open [issue](https://github.com/prisma/prisma/issues/5220) to prevent throwing errors when passing additional fields. There is a [workaround](https://github.com/prisma/prisma/issues/5220#issuecomment-1262255731) suggested by a user and another one [here](https://github.com/prisma/prisma/issues/5220#issuecomment-1445576235).

**Reply by MrGigabit** [reply]
Appreciate it!

**Reply by ludralph** [reply]
You're welcome 🙏
